osdep/subprocess-posix.c

```diff
@@ -74,13 +80,13 @@ static int as_execvpe(const char *path, const char *file, char *const argv[],
 // signal handlers first so nothing funny happens.
 static void reset_signals_child(void)
 {
-    struct sigaction sa = { 0 };
+    struct sigaction sa = { .sa_handler = SIG_DFL };
     sigset_t sigmask;
-    sa.sa_handler = SIG_DFL;
-    sigemptyset(&sigmask);
 
-    for (int nr = 1; nr < SIGNAL_MAX; nr++)
+    for (int nr = SIGNAL_MIN; nr < SIGNAL_MAX; nr++)
```

Comment:
this would cause problem if there are signal handlers left configured, which is the whole point of the function

Reply:
In `man sigaction`:

> The glibc wrapper function for sigaction() gives an error (EINVAL) on attempts to change the disposition of  the  two  real-time signals used internally by the NPTL threading implementation.  See nptl(7) for details.

So unless we call the syscall directly, we have no way to change these two signals.

Reply:
The only way to strictly achieve async-signal-safety is to eliminate the use of `fork`, which is done in this PR.